runsc/container/container_test.go

```diff
@@ -3127,3 +3129,160 @@ func TestOverlayByMountAnnotation(t *testing.T) {
+	}
+	for i := 0; i < 1000; i++ {
+		// Create a random regular file.
+		tempFile, err := ioutil.TempFile(bigDir, "erofs_test_")
+		if err != nil {
+			t.Fatalf("ioutil.TempFile() failed: %v", err)
+		}
+		// Write some random data into the regular file.
+		if _, err := fmt.Fprint(tempFile, rand.Int()); err != nil {
+			t.Fatalf("fmt.Fprint() failed: %v", err)
+		}
+		name := tempFile.Name()
+		// Create a symlink that points to the regular file.
+		if err := os.Symlink(name, name+"@"); err != nil {
+			t.Fatalf("os.Symlink() failed: %v", err)
+		}
+		tempFile.Close()
+	}
+
+	// Create a big file.
+	bigFile, err := ioutil.TempFile(sourceDir, "erofs_test_")
+	if err != nil {
+		t.Fatalf("ioutil.TempFile() failed: %v", err)
+	}
+	for i := 0; i < 5000; i++ {
+		if _, err := fmt.Fprint(bigFile, rand.Int()); err != nil {
+			t.Fatalf("fmt.Fprint() failed: %v", err)
+		}
+	}
+	bigFile.Close()
+
+	// Create a test script which can be used to get the checksums
+	// from a specified directory.
+	scriptFile := path.Join(sourceDir, "test-script")
+	f, err := os.OpenFile(scriptFile, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		t.Fatalf("os.OpenFile() failed: %v", err)
+	}
+	if _, err := f.Write([]byte(`#!/bin/bash
+set -u -e -o pipefail
+dir=$1
+find $dir -printf "%P\n" | sort | md5sum
+find $dir -type f | sort | xargs cat | md5sum
+find $dir -type l | sort | xargs readlink | md5sum`)); err != nil {
+		t.Fatalf("Write() failed: %v", err)
+	}
+	f.Close()
+
+	// Create the EROFS images in a temporary directory.
+	imageDir, err := ioutil.TempDir(testutil.TmpDir(), "erofs-images")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir() failed: %v", err)
+	}
+
+	images := []struct {
+		name string
+		cmd  string
+	}{
+		{
+			// Generate extended inodes. Inline regular files if possible.
+			name: "image1",
+			cmd:  "mkfs.erofs -E force-inode-extended",
+		},
+		{
+			// Generate extended inodes. Do not inline regular files.
+			name: "image2",
+			cmd:  "mkfs.erofs -E force-inode-extended -E noinline_data",
+		},
+		{
+			// Generate compact inodes. Inline regular files if possible.
+			name: "image3",
+			cmd:  "mkfs.erofs -E force-inode-compact",
+		},
+		{
+			// Generate compact inodes. Do not inline regular files.
+			name: "image4",
+			cmd:  "mkfs.erofs -E force-inode-compact -E noinline_data",
+		},
+	}
+
+	for _, i := range images {
+		cmd := fmt.Sprintf("%s %s %s", i.cmd, path.Join(imageDir, i.name), sourceDir)
+		if out, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput(); err != nil {
+			t.Fatalf("exec: sh -c %q, err: %v, out: %s", cmd, err, out)
+		}
+	}
+
+	var checksums string
+
+	// Get the checksums from the source directory on host.
+	if out, err := exec.Command(scriptFile, sourceDir).CombinedOutput(); err != nil {
+		t.Fatalf("exec: %s %s, err: %v, out: %s", scriptFile, sourceDir, err, out)
+	} else {
+		checksums = string(out)
+	}
+
+	for name, conf := range configs(t, false /* noOverlay */) {
```

Comment:
This will generate a bunch of dimensions that may not be relevant to EROFS (overlay x platform). I think it should be okay to just use 1 config from `testutil.TestConfig(t)`. We are already adding a dimension for the image (compact/extended x inline), so the test is already long-ish.

Reply:
Make sense. Will do.